### ISSUE TITLE
Race condition on creation of clusters 

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe-cluster-testbench#889]
We have now reappearing cluster creation failures, due to a race condition when creating the generation and updating the related channel.

The symptom we observe is that there is an incident in testbench that a cluster can't be created. 

A potential reason is that the channel update fails, because the generation is not yet created. But this is not transparent from outside.

[In testbench logs we see something like this, when cluster creation fails](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22zeebe-io%22%0Aresource.labels.location%3D%22europe-west1-b%22%0Aresource.labels.cluster_name%3D%22zeebe-cluster%22%0Aresource.labels.namespace_name%3D%22testbench-1-x-prod%22%0Alabels.k8s-pod%2Fapp%3D%22testbench%22;pinnedLogId=2023-09-14T09:41:46.053105684Z%2Ff3opoemayigfe95i;cursorTimestamp=2023-09-14T09:41:46.053105684Z;duration=PT15M?project=zeebe-io)

```
Expected to create cluster compensation-exclusive-transaction, but failed

jakarta.ws.rs.client.ResponseProcessingException: java.io.IOException: BAD REQUEST returned from URI: https://api.cloud.ultrawombat.com/clusters/, requestBody:{"name":"compensation-exclusive-transaction","planTypeId":"0282d9dd-266f-4db6-a451-724ff62648f4","channelId":"cf355219-cc2d-4266-83e1-2430998ddf30","generationId":"9728580a-744c-4bb4-a771-e4dc3b859c49","regionId":"cab01f7b-0b95-42a3-83d0-b66d53fd5255"}, responseBody: Generation is not part of allowed generations of selected channel.
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.filterResponse(ClientInvocation.java:671) ~[testbench.jar:3.0.5-SNAPSHOT]
	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.invoke(ClientInvocation.java:428) ~[testbench.jar:3.0.5-SNAPSHOT]
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:134) ~[testbench.jar:3.0.5-SNAPSHOT]
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:103) ~[testbench.jar:3.0.5-SNAPSHOT]
	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:61) ~[testbench.jar:3.0.5-SNAPSHOT]
	at jdk.proxy2.$Proxy42.createCluster(Unknown Source) ~[?:?]
	at io.zeebe.clustertestbench.handler.CreateClusterInCamundaCloudHandler.handle(CreateClusterInCamundaCloudHandler.java:35) ~[testbench.jar:3.0.5-SNAPSHOT]
	at io.zeebe.clustertestbench.handler.JobHandlerWithEnrichedLogger.handle(JobHandlerWithEnrichedLogger.java:26) ~[testbench.jar:3.0.5-SNAPSHOT]
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.executeJob(JobRunnableFactory.java:44) ~[testbench.jar:3.0.5-SNAPSHOT]
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactory.lambda$create$0(JobRunnableFactory.java:39) ~[testbench.jar:3.0.5-SNAPSHOT]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
	at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.io.IOException: BAD REQUEST returned from URI: https://api.cloud.ultrawombat.com/clusters/, requestBody:{"name":"compensation-exclusive-transaction","planTypeId":"0282d9dd-266f-4db6-a451-724ff62648f4","channelId":"cf355219-cc2d-4266-83e1-2430998ddf30","generationId":"9728580a-7…[message truncated due to size]
```

## Workaround

If you run into this issue see the following checklist to overcome this:

[How to fix the QA run if we run into timeouts with generation missing:](https://camunda.slack.com/archives/C013MEVQ4M9/p1692017716010219?thread_ts=1691998093.226669&cid=C013MEVQ4M9)
* Check out whether you find the generation in accounts https://accounts.cloud.ultrawombat.com/consoleadmin/generations
* Find out the name for the uuid.
* Update the internal channel https://accounts.cloud.ultrawombat.com/consoleadmin/channels (add the generation, to make it available)
* Retry the PI which had an incident. Enjoy.

**Comment by Zelldon** [comment]
What I did so far:

* I investigated the code, but right now it is unclear to me why this can happen. To me it seems that we always for the response, before we continue. Needs to be verified for methods with void as return value (might be here not the case).
* I added some more logging, so we can better see and observe if something fails. https://github.com/zeebe-io/zeebe-cluster-testbench/pull/892 and https://github.com/zeebe-io/zeebe-cluster-testbench/pull/891
